packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -209,6 +211,11 @@ class InkSparkle extends InteractiveInkFeature {
     _turbulenceSeed = turbulenceSeed ?? math.Random().nextDouble() * 1000.0;
   }
 
+  void _handleStatusChanged(AnimationStatus status) {
+    if (status == AnimationStatus.completed)
+      dispose();
```

Comment:
can we test this?

For example, add some kind of `debugHasFeature` API on `MaterialInkController` and assert in a test that once the animation is done the controller doesn't contain this feature anymore?

Reply:
MaterialInkController is often implemented, so adding `debugHasFeature ` may be breaking

Reply:
behold

Reply:
Thanks I hate it